cluster-autoscaler/core/test/common.go

```diff
@@ -239,6 +241,25 @@ func NewScaleTestAutoscalingContext(
 	}, nil
 }
 
+// MockAsyncNodeGroupStateChecker is a mock AsyncNodeGroupStateChecker to be used in tests
+type MockAsyncNodeGroupStateChecker struct {
+	T                   *testing.T
+	IsUpcomingNodeGroup map[string]bool
+}
+
+// IsUpcoming simulates checking if node group is upcoming.
+func (p *MockAsyncNodeGroupStateChecker) IsUpcoming(nodeGroup cloudprovider.NodeGroup) bool {
+	val, exists := p.IsUpcomingNodeGroup[nodeGroup.Id()]
```

Comment:
nit: this function could just `return p.IsUpcomingNodeGroup[nodeGroup.Id()]`, since the default value of bool is `false`.